blog/2024-11-20-decompressing-work/index.mdx

```diff
@@ -0,0 +1,211 @@
+
+```shell-session
+etcdserver: request is too large
+```
+
+> You may see an error message like `Request entity too large: limit is
+> 3145728`; that means you are hitting the Kubernetes API limit, not the etcd
+> one.
+
+While it takes a lot of YAML to be over 1.5MB, you can easily reach such a limit
+in your Promise. The [Prometheus
+Operator](https://github.com/prometheus-operator/prometheus-operator), for
+example, includes 3.7MB of YAML!
+
+This brings us back to the [change introduced by
+243](https://github.com/syntasso/kratix/pull/243). In this update, we introduced
+gzip compression for the Workload contents before persisting the Work into etcd.
+This significantly reduces the size of the Work object (gzip documentation
+mentions an average of [70% reduction in
+size](https://www.gnu.org/software/gzip/manual/gzip.html). For the Prometheus
+Operator, for example, the size of the Work object went from 3.7MB to about
+490KB—an 87% reduction 🎉!
+
+The downside? If you inspect the Work object, you’ll see base64-encoded binary
+data instead of some nice to read YAML.
+
+<figure className="diagram">
+  <img className="large" src={Figure03} alt="Screenshot of a terminal showing
+  the Work object with binary data in the contents of a Workload" />
+
+  <figcaption>A compressed Work Object</figcaption>
+</figure>
+
+You can still read it though. To inspect a workload’s contents, decode the
+base64 data, then unzip it using this command:
+
+```shell-session
+kubectl get work <work-name> \
+  -o jsonpath='{.spec.workloadGroups[0].workloads[0].content}' \
+  | base64 -d \
+  | gzip -d
+```
+
+<figure className="diagram">
+  <img className="large" src={Figure04} alt="Screenshot of a terminal showing
+  the decompressed contents of a workload from a Work object" />
+
+  <figcaption>A decompressed workload</figcaption>
+</figure>
+
+:::tip
+
+Check our [Troubleshooting guide](/main/troubleshooting) for more information on
+how to debug Kratix, including inspecting Works and WorkPlacements.
+
+:::
+
+Despite compression, large Work objects may still pose challenges. While this
+update provides temporary relief, we’ll need to revisit the structure to allow
+users to have unbounded fun with their Promises. But that’s a story for another
+day.
+
+## Conclusion
+
+In this post, we've dived into the internals of Kratix to understand how a
+document moves from the Platform Cluster to a Destination. We've seen how the
+Work and WorkPlacement objects are used to schedule and write documents to the
+State Store. We've also seen how the recent change to compress the Workload
+contents has helped reduce the size of the Work object.
+
+We hope this post has given you a better understanding of how Kratix works under
+the hood. If you have any questions or feedback (or want to see more blog posts
+like this) please don't hesitate to reach out to us on
+[Slack](https://kratixworkspace.slack.com/) or
+[GitHub](https://github.com/syntasso/kratix).
```

Comment:
I noticed there is a LOT of white space at the bottom of the blog page when i spun it up locally 👀 
![Screenshot 2024-11-22 at 10 16 59](https://github.com/user-attachments/assets/5fd3729d-f3b2-4638-9790-081aa899c77b)

Reply:
i think thats you... 🤠

![image](https://github.com/user-attachments/assets/e4241993-0cd8-4018-a72e-387c3ce34550)